app/ota_client.py

```diff
@@ -896,48 +887,151 @@ def _gen_regular_files(self, rootfs_dir, regulars_file, target_dir):
         """
         self._boot_vmlinuz = None  # clear
         self._boot_initrd = None  # clear
-        with tempfile.NamedTemporaryFile(delete=False) as flog:
-            log_file = flog.name
-            with open(flog.name, "w") as fl:
-                res = True
-                with open(regulars_file) as f:
-                    try:
-                        cwd = os.getcwd()
-                        os.chdir(target_dir)
-                        logger.debug(f"target_dir: {target_dir}")
-                        prev_inf = ""
-                        rlist = []
-                        for l in f.readlines():
-                            rlist.append(RegularInf(l))
-                        sorted_rlist = sorted(rlist, key=lambda x: x.sha256hash)
-                        # with open('./tests/sorted_regular_list.txt', "w") as fdst:
-                        #    for lst in sorted_list:
-                        #        fdst.write(str(lst) + '\n')
-                        for l in sorted_rlist:
-                            regular_inf = l
-                            logger.debug(
-                                f"file: {regular_inf.path}, hash: {regular_inf.sha256hash}"
+
+        cwd = os.getcwd()
+        os.chdir(target_dir)
+
+        rfiles_list = []
+        with open(regulars_file) as f:
+            logger.debug(f"target_dir: {target_dir}")
+            rfiles_list = [RegularInf(l) for l in f.readlines()]
+
+        # process all regular files here
+        try:
+            self._process_regular_files(rootfs_dir, rfiles_list, target_dir)
+        except:
+            logger.exception(f"gen regular files error!")
+            return False
+        finally:
+            os.chdir(cwd)
+
+        return True
+
+    def _process_regular_files_pool_init(self, gvar_dict):
+        """
+        Used by _process_regular_files
+        Init the worker pool with shared variables
+
+        DO NOT call this method from other functions
+        except for _process_regular_files
+        """
+        global global_var_dict
+        global_var_dict = gvar_dict
+
+    def _process_regular_files_exit(self, gvar_dict):
+        """
+        Used by _process_regular_files
+        Update corresponding class attributes
+
+        DO NOT call this method from other functions
+        except for _process_regular_files
+        """
+        for k, v in gvar_dict.items():
```

Comment:
About this block, I think 
setattr(self, "rollback_dict", gvar_dict["staging-dict-_rollback_dict"])
since other function already knows 
- gvar_dict["staging-dict-_rollback_dict"] is dict-type
- having "staging-dict-_rollback_dict" key
I like simple implementation.

Reply:
Sure, currently directly setattr is enough.